### DISCUSSION TITLE
`lockFileMaintenance` only for a single `package-lock.json`

### DISCUSSION THREAD

**Discussion by pmwmedia** [renovatebot/renovate#18936] [Request Help]
I would like to apply `lockFileMaintenance` only for a single `package-lock.json` file in the root directory of my repository. The reason is that the other `package-lock.json` files have dependencies to generated OpenAPI packages, which are only present when running a Maven command before `lockFileMaintenance`. As renovate cannot execute any Maven commands, I have to exclude all related `package-lock.json` files.

How can I have `lockFileMaintenance` for only a single `package-lock.json` file?

```json5
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [ "config:base" ],
    "lockFileMaintenance": {
        "enabled": true,
        "automerge": true,
        "automergeType": "branch",
        "commitMessageAction": "Update package lock file"
        // Update "<root>/package-lock.json`, but ignore "<root>/<sub-folders>/package-lock.json`
    }
}
```

**Comment by rarkins** [comment]
I'm not sure if the following will work but please try it:

```
{
  "packageRules": [{
    "matchFiles": ["package.json"],
    "lockFileMaintenance": {
        "enabled": true
    }
  }]
}
```

**Reply by pmwmedia** [reply]
Unfortunately, this disables lock maintenance completely. Lock file maintenance has even disappeared from the dependency dashboard.

**Reply by rarkins** [reply]
In this case I'm not sure we have this capability today. Please create a reproduction repo (eg with two lock files, one of which you don't want to update) and then create a feature request issue

**Reply by pmwmedia** [reply]
I have just created a feature request: #18966
